README.md

```diff
@@ -1 +1,78 @@
-# platform-backend
\ No newline at end of file
+# platform-backend
+
+A Go-based API server built using the `Gin` framework. It serves as the backend for a serverless platform, which makes consuming `Capps` more easily. It is part of the[`rcs-ocm-deployer`](https://github.com/dana-team/rcs-ocm-deployer) and [`container-app-operator`](https://github.com/dana-team/container-app-operator) ecosystem, and is a complementary project to the [`platform-frontend`](https://github.com/dana-team/platform-frontend) project.
+
+## Documentation
+
+The API is documented at the [docs directory](./docs/api) on this repo.
+
+## Getting Started
+
+### OAuth
+
+The API uses the built-in `OAuth` server of `OpenShift` in order to auhenticate users.
+
+#### Environment Variables
+
+To use the `OpenShift OAuth` and run the code, create an `.env` file on your local machine. The `.env` file should include the following environment variables:
+
+```bash
+CLUSTER_NAME=ocp-rcs-example    # change to match your own
+CLUSTER_DOMAIN=dana.com         # change to match your own
+KUBE_AUTH_BASE_URL="https://oauth-openshift.apps.${CLUSTER_NAME}.${CLUSTER_DOMAIN}"
+KUBE_API_BASE_URL="https://api.${CLUSTER_NAME}.${CLUSTER_DOMAIN}:6443"
+
+INSECURE_SKIP_VERIFY=true
+KUBE_CLIENT_ID="openshift-challenging-client"
+KUBE_AUTH_URL="${KUBE_AUTH_BASE_URL}/oauth/authorize"
+KUBE_TOKEN_URL="${KUBE_AUTH_BASE_URL}/oauth/token"
+KUBE_USERINFO_URL="${KUBE_API_BASE_URL}/apis/user.openshift.io/v1/users/~"
+KUBE_API_SERVER="${KUBE_API_BASE_URL}"
+
+cat <<EOF > .env
+INSECURE_SKIP_VERIFY=${INSECURE_SKIP_VERIFY}
+KUBE_CLIENT_ID=${KUBE_CLIENT_ID}
+KUBE_AUTH_URL=${KUBE_AUTH_URL}
+KUBE_TOKEN_URL=${KUBE_TOKEN_URL}
+KUBE_USERINFO_URL=${KUBE_USERINFO_URL}
+KUBE_API_SERVER=${KUBE_API_SERVER}
+EOF
+```
+
+### Run the server
+
+Use the following `Makefile` target in order to run the server:
+
+```
+$ make run
+```
+
+### Routes examples
+
+#### Login
+
+In order to `login` use the `/v1/login/` route in the following way. You should receieve a `token` in return:
+
+```bash
+$ curl -X POST -H "Content-Type: application/json" -d '{"username": "<USERNAME>","password": "<PASSWORD>"}' "localhost:8080/v1/login/"
+
+# response
+{"token":"<TOKEN>"}
+```
+
+#### Get Namespaces
+
+Use the `/v1/namespaces/` route to get all the namespaces in the cluster:
+
+```bash
+$ curl -H "Authorization: Bearer <TOKEN>" "localhost:8080/v1/namespaces/"
+
+# response
+{"namespaces":[{"name":"project1"},{"name":"project2"}],"count":2}
+```
+
+## Build an image
```

Comment:
This section isn't clear. As an user, I want to see how I can build and run it using Golang and also how to build and run it as Docker user.

Reply:
I'd remove this section, or add it to where you suggest to run `make run` (so we can consider it as an alternative to `make run` - to run it in the Docker container).